qcelemental/models/molecule.py

```diff
@@ -186,25 +194,27 @@ class Molecule(ProtoModel):
         )
     fragments: Optional[List[Array[np.int32]]] = Schema(
         None,
-        description="An indication of which sets of atoms are fragments within the Molecule. This is a list "
-                    "of 1-D array-like objects. Each entry in the list indicates a new fragment. The index "
-                    "of this list matches the index of ``fragment_charges`` and ``fragment_multiplicities``. "
+        description="An indication of which sets of atoms are fragments within the Molecule. This is a list of shape "
+                    "(NFrag) of 1-D array-like objects of arbitrary shape. Each entry in the list indicates a new "
+                    "fragment. The index "
+                    "of the list matches the index of ``fragment_charges`` and ``fragment_multiplicities``. "
                     "The 1-D array-like objects are sets of atom indices indicating the atoms which compose "
                     "the fragment. The atom indices match the index of all other per-atom settings like ``symbols`` "
                     "and ``real``."
     )
     fragment_charges: Optional[List[float]] = Schema(
         None,
-        description="The total charge of each fragment in the ``fragments`` list. The index of this list matches the "
-                    "index of ``fragment`` list. This must be provided if the ``molecular_charge`` is not 0, otherwise "
-                    "it is assumed fragments are all 0 charge as well (only a condition if there are ``fragments``)."
+        description="The total charge of each fragment in the ``fragments`` list of shape (NFrag). The index of this "
+                    "list matches the index of ``fragment`` list. This must be provided if the ``molecular_charge`` is "
```

Comment:
no on the "must be provided" here and below (and maybe on molecular_chg/mult?). Validation will fill in frag chgmult and switch mult to open shell if the system demands it according to [rules](https://github.com/MolSSI/QCElemental/blob/master/qcelemental/molparse/chgmult.py#L103-L134)